### PR TITLE
Tantivy-0.19.2 updates

### PR DESCRIPTION
**Pull request by Sidhant29**
This PR rebases the master from `Kapiche/tantivy-py` on `quickwit/tantivy-py` with the updates from tantivy version 0.19.2,
From the following commit:
https://github.com/quickwit-oss/tantivy-py/commit/164adc87e1a033117001cf70e38c82a53014d985

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -818,9 +818,8 @@ dependencies = [
 
 [[package]]
 name = "pyo3-macros-backend"
-version = "0.18.0"
```
Comment (Sidhant29) [inline Cargo.lock:821]:
```suggestion
version = "0.18.1"
```

File: Cargo.lock [hunk 2]
```diff
@@ -818,9 +818,8 @@ dependencies = [
 
 [[package]]
 name = "pyo3-macros-backend"
```
Comment (Sidhant29) [inline Cargo.lock:820]:
```suggestion
name = "pyo3-macros-backend"
version = "0.18.1"
```

File: Makefile [hunk 3]
```diff
@@ -6,15 +6,24 @@ endif
...
 
 format:
-	rustfmt src/*.rs
+	cargo fmt
+
+build:
+	maturin build --interpreter python3.7 python3.8 python3.9 python3.10 python3.11
+
+build:
+	maturin build --interpreter python3.7 python3.8 python3.9 python3.10 python3.11
```
Comment (cjrh) [inline Makefile:26]: Why so many of these? Looks like there are 3

**Review by cjrh — APPROVED** [review]
I am approving but please see my comment. It seems to me that there is some kind of merge error in the Makefile. After this is fixed it is fine to merge. I didn't check whether upstream had this also.

**Review by wallies — CHANGES_REQUESTED** [review]
Just the duplicates in the Makefile to fix then it looks good

File: Makefile (hunk 3 above)
Comment (Sidhant29) [inline Makefile:26]:
```suggestion
```

**Review by wallies — APPROVED** [review]

**Pull request merged**